tmt/steps/provision/__init__.py

```diff
@@ -316,16 +316,93 @@ def _socket_path_hash(
     return None
 
 
-# Default rsync options
-DEFAULT_RSYNC_OPTIONS = ["-s", "-R", "-r", "-z", "--links", "--safe-links", "--delete"]
-
-DEFAULT_RSYNC_PUSH_OPTIONS = ["-s", "-R", "-r", "-z", "--links", "--safe-links", "--delete"]
-DEFAULT_RSYNC_PULL_OPTIONS = ["-s", "-R", "-r", "-z", "--links", "--safe-links", "--protect-args"]
-
 #: A pattern to extract ``btime`` from ``/proc/stat`` file.
 STAT_BTIME_PATTERN = re.compile(r'btime\s+(\d+)')
 
 
+@container
+class TransferOptions:
+    """Options for transferring files to/from the guest."""
+
+    #: Apply permissions to the destination files
+    chmod: Optional[int] = None
+
+    #: Enable compression during transfer
+    compress: bool = False
+
+    #: Delete extraneous files from destination directory
+    delete: bool = False
+
+    #: Exclude files matching any of these patterns
+    exclude: list[str] = field(default_factory=list, normalize=tmt.utils.normalize_string_list)
+
+    #: Copy symlinks as symlinks
+    links: bool = False
+
+    #: Preserve file permissions
+    preserve_perms: bool = False
+
+    #: Protect file and directory names from interpretation
+    protect_args: bool = False
+
+    #: Recurse into directories
+    recursive: bool = False
+
+    #: Use relative paths
+    relative: bool = False
+
+    #: Ignore symlinks that point outside the source tree
+    safe_links: bool = False
+
+    def to_rsync(self) -> list[str]:
+        """Convert to rsync command line options."""
+
+        options = []
+
+        if self.chmod:
+            options.append(f'--chmod={oct(self.chmod)[2:]}')
+        if self.compress:
+            options.append('-z')
+        if self.delete:
+            options.append('--delete')
+        if self.exclude:
+            options += [f'--exclude {pattern}' for pattern in self.exclude]
+        if self.links:
+            options.append('--links')
+        if self.protect_args:
+            options.append('-s')
+        if self.preserve_perms:
+            options.append('-p')
+        if self.recursive:
+            options.append('-r')
+        if self.relative:
+            options.append('-R')
+        if self.safe_links:
+            options.append('--safe-links')
+
+        return options
+
+
+DEFAULT_PUSH_OPTIONS = TransferOptions(
+    protect_args=True,
+    relative=True,
+    recursive=True,
+    compress=True,
+    links=True,
+    safe_links=True,
+    delete=True,
+)
+
+DEFAULT_PULL_OPTIONS = TransferOptions(
+    protect_args=True,
+    relative=True,
+    recursive=True,
+    compress=True,
+    links=True,
+    safe_links=True,
+)
```

Comment:
Quite a few `=True` - would it make sense to make `True` the default for these options?